### ISSUE TITLE
Two defects in socket.scala

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#2295]
I noticed the following whilst working on a PR which touches socket.scala.

I can not fix all the defects at once, so I capture the info here.

```
type sockaddr =
    CStruct2[sa_family_t,        // sa_family                                   
             CArray[CChar, _14]] // sa_data, size = 14 in OS X and Linux        
  type sockaddr_storage = CStruct1[sa_family_t] // ss_family                    

```
1) The declaration of sockaddr_storage is a complaint/bug_report waiting to happen.
    Using the Open Group 2018 socket.h description, both the size (2 bytes, should be 16)
    and the alignment of the type are wrong. 

2) It is not a requirement of the spec, but since the sockaddr usually contains pointers,
    the declaration should be the same size, but done in a way which declares a Ptr,
    This should give the entire structure 8 byte alignment.  Not a big win for IPv4,
    but IPv6 addresses are, for runtime efficiency, designed to be 8 byte aligned.

**Comment by LeeTibbert** [comment]
Both of these trigger the need for a change in `socket.h`, where I first discovered the defects.

**Comment by LeeTibbert** [comment]
Relevant to the discussion of alignment above.  I believe that alloc, and possibly stackalloc, align returned
memory to 8-byte boundaries (__long__ and obscure story).  Consider that when chasing number 2 above.
It might reduce the urgency/concern.

**Comment by LeeTibbert** [comment]
resolved by merged PR #2630
